### PR TITLE
Remove master documentation from being indexable by search engines

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58092 Adds an aten::_ops namespace with unambiguous function names
* **#58056 Remove master documentation from being indexable by search engines**

This PR addresses an action item in #3428: disabling search engine
indexing of master documentation. This is desireable because we want to
direct users to our stable documentation (instead of master
documentation) because they are more likely to have a stable version of
PyTorch installed.

Test Plan:
1. run `make html`, check that the noindex tags are there
2. run `make html-stable, check that the noindex tags aren't there

Differential Revision: [D28490504](https://our.internmc.facebook.com/intern/diff/D28490504)

### CODE REVIEW

**Comment by mattip** [comment]
`stable` is a symlink to the "current version", which should be produced by tagging a version.  Committing a tag should produce a version of the documents for that tag, but I have not seen this work in practice. I want to carefully track what happens this week when a version is tagged to see why the CI is not running the scripts as designed. Once that version becomes the release, the symlink is changed to point at that version.

In short: I think this should work as desired.

**Comment by zou3519** [comment]
@mattip that sounds good to me. Should we wait until you've tracked exactly what happens during releases before merging this?

**Comment by mattip** [comment]
> Should we wait 

I think not, this will definitely improve the nightly builds of master, and it will be easy enough to remove `<meta name="robots" content="noindex">` from the stable documentation if needed.

**Review by brianjo — APPROVED** [review]

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28490504).
